rpm/SPECS/docker-ce.spec

```diff
@@ -34,15 +34,13 @@ BuildRequires: cmake
 BuildRequires: gcc
 BuildRequires: git
 BuildRequires: glibc-static
-BuildRequires: libarchive
 BuildRequires: libtool
 BuildRequires: libtool-ltdl-devel
 BuildRequires: make
 BuildRequires: pkgconfig
 BuildRequires: pkgconfig(systemd)
 BuildRequires: systemd-devel
 BuildRequires: tar
-BuildRequires: which
```

Comment:
This one was originally added in https://github.com/docker/docker-ce-packaging/commit/fda74c4df83554ae2662184b4a5a1a1eaaf6927a (through https://github.com/docker/docker-ce-packaging/pull/314).


My _guess_ is that some script at some point incorrectly used `which <some command>` instead of `command -v <some command>`, but I couldn't find a reference to it either at the commit that added it, nor at current master, so I'm gonna assume here that it was never needed at all;


```bash
git rev-parse --verify HEAD
fda74c4df83554ae2662184b4a5a1a1eaaf6927a

git grep 'which '
deb/common/docker-ce.docker.init:       # see also init_is_upstart in /lib/lsb/init-functions (which isn't available in Ubuntu 12.04, or we'd use it directly)
deb/gen-deb-ver:    # be compared to determine which happened later, the commit hash identifes
rpm/gen-rpm-ver:    # be compared to determine which happened later, the commit hash identifes
static/gen-static-ver:    # be compared to determine which happened later, the commit hash identifes

git grep 'command '
deb/common/docker-ce.postinst:          echo "You can use the 'docker engine update' command to update your system, or switch to using the Enterprise packages."
deb/common/rules:       dh $@ --with=bash-completion $(shell command -v dh_systemd_enable > /dev/null 2>&1 && echo --with=systemd)
deb/gen-deb-ver:    # as a pre-release before version v0.0.0, so that the go command prefers any
rpm/gen-rpm-ver:    # as a pre-release before version v0.0.0, so that the go command prefers any
static/gen-static-ver:    # as a pre-release before version v0.0.0, so that the go command prefers any


git checkout master

git rev-parse --verify HEAD
2f0c865a12578226abe568ca772c9ca06e58516b

git grep 'which '
Jenkinsfile:        // Otherwise it accidentally pulls armel images which then breaks the verify step
Jenkinsfile:                // in Jenkins' BlueOcean view, which truncates names....
Jenkinsfile:                // in Jenkins' BlueOcean view, which truncates names....
Jenkinsfile:                // in Jenkins' BlueOcean view, which truncates names....
Jenkinsfile:                // in Jenkins' BlueOcean view, which truncates names....
LICENSE:      form, that is based on (or derived from) the Work and for which the
LICENSE:      with the Work to which such Contribution(s) was submitted. If You
README.md:which can be found in the "rpm" and "deb" subdirectories, as well as scripts to
deb/build-deb:# - pkgRevision (usually "-0", see above), which allows updating packages with
deb/build-deb:#   we prefix the codename with a tilde (~), which effectively excludes it from
deb/build-deb:# EPOCH because of our use of CalVer, which made version comparing not work in
deb/common/rules:# zstd compression, which is non-standard, and breaks 'dpkg-sig --verify'
deb/gen-deb-ver:# non-pre-release versions, which would prevent users from updating from a pre-
deb/gen-deb-ver:        # be compared to determine which happened later, the commit hash identifes
deb/gen-deb-ver:                # Using BSD date (macOS), which doesn't support the --date option
rpm/SPECS/docker-ce.spec:# Provides modprobe, which we use to load br_netfilter if not loaded.
rpm/gen-rpm-ver:# - Version: > 1 : Only to be used for packaging-only changes (new package built for a version for which a package was already built/released)
rpm/gen-rpm-ver:# in this field, which is still allowed, but considered deprecated; see
rpm/gen-rpm-ver:# non-pre-release versions, which would prevent users from updating from a pre-
rpm/gen-rpm-ver:# > The tilde symbol (‘~’) is used before a version component which must sort
rpm/gen-rpm-ver:# > which wouldn’t otherwise sort appropriately.
rpm/gen-rpm-ver:        # be compared to determine which happened later, the commit hash identifes
rpm/gen-rpm-ver:                # Using BSD date (macOS), which doesn't support the --date option
static/gen-static-ver:  # be compared to determine which happened later, the commit hash identifes
static/gen-static-ver:          # Using BSD date (macOS), which doesn't support the --date option
verify:                         # may still be using "rawhide", which is not present on our package

git grep 'command '
deb/gen-deb-ver:        # as a pre-release before version v0.0.0, so that the go command prefers any
install-containerd-helpers:     if command -v dnf5; then
install-containerd-helpers:     elif command -v dnf; then
rpm/gen-rpm-ver:        # as a pre-release before version v0.0.0, so that the go command prefers any
static/gen-static-ver:  # as a pre-release before version v0.0.0, so that the go command prefers any
```